tethys_dataset_services/engines/geoserver_engine.py

```diff
@@ -1238,84 +1374,117 @@ def add_table_to_postgis_store(self, store_id, table, debug=False):
-                geometry_type='LineString',
-                geometry_srid=32144,
-                default_style_id='my_workspace:pipes',
-                debug=True
+        retries_remaining = 3
+        while retries_remaining > 0:
+            response = requests.post(
+                url,
+                headers=headers,
+                auth=(self.username, self.password),
+                data=xml,
             )
 
-        """  # noqa: E501, W605
-        # Process identifier
-        store_workspace_name, store_name = self._process_identifier(postgis_store_id)
-
-        # Get Existing PostGIS Store
-        store = self.catalog.get_store(store_name, workspace=store_workspace_name)
-
-        # Define virtual table / sql view
-        epsg_code = 'EPSG:{0}'.format(geometry_srid)
-        geometry = JDBCVirtualTableGeometry(geometry_column, geometry_type, str(geometry_srid))
-
-        if parameters is not None:
-            jdbc_parameters = []
-            for parameter_args in parameters:
-                jdbc_parameters.append(JDBCVirtualTableParam(*parameter_args))
-            parameters = jdbc_parameters
-
-        sql_view = JDBCVirtualTable(feature_type_name, sql, 'false', geometry, key_column, parameters)
+            # Raise an exception if status code is not what we expect
+            if response.status_code == 201:
+                log.info('Successfully created featuretype {}'.format(feature_name))
+                break
+            if response.status_code == 500 and 'already exists' in response.text:
+                break
+            else:
+                retries_remaining -= 1
+                if retries_remaining == 0:
+                    msg = "Create Feature Type Status Code {0}: {1}".format(response.status_code, response.text)
+                    exception = requests.RequestException(msg, response=response)
+                    log.error(exception)
+                    raise exception
+
+        # Add styles to new layer
+        self.update_layer_styles(
+            layer_id=layer_id,
+            default_style=default_style,
+            other_styles=other_styles
+        )
 
-        # Publish Feature Type
-        self.catalog.publish_featuretype(feature_type_name, store, epsg_code, jdbc_virtual_table=sql_view)
+        # GeoWebCache Settings
+        gwc_layer_path = os.path.join(self.XML_PATH, 'gwc_layer_template.xml')
+        url = self.get_gwc_endpoint(public=False) + 'layers/' + workspace + ':' + feature_name + '.xml'
+        headers = {
+            "Content-type": "text/xml"
+        }
+        with open(gwc_layer_path, 'r') as gwc_layer_file:
+            text = gwc_layer_file.read()
+            template = Template(text)
+            xml = template.render(context)
 
-        # Wrap Up
-        r_feature_layer = self.catalog.get_layer(feature_type_name)
+        retries_remaining = 300
+        while retries_remaining > 0:
+            response = requests.post(
```

Comment:
GSAPI version uses `put`. Why change to `post`?

Reply:
This was a bug that Tran had shown me a while ago. The put call returned a warning from geoserver literally saying "use post".